packages/input_schema/src/input_schema.ts

```diff
@@ -138,16 +153,16 @@ export function validateExistenceOfRequiredFields<T extends Record<string, any>>
  *
  * This way we get the most accurate error message for user.
  */
-export function validateInputSchema<T extends Record<string, any>>(validator: Ajv, inputSchema: T) {
+export function validateInputSchema(validator: Ajv, inputSchema: InputSchemaUnchecked): asserts inputSchema is InputSchema {
```

Comment:
Is the signature change here correct? I think you should be able to pass any object as `inputSchema` value, otherwise this would be a breaking change.

Reply:
Yeah, now when I think about it....
Yes, I thought that the deep partial type would be enough, but you are correct, since type would fail if you pass for example `description: 3`

Reply:
I changed it so unchecked type is only one for middle state and everything else is `Record<string, unknown>` or `Record<string<Record<string, unknown>>`